webui/react/src/pages/Cluster/ClusterHistoricalUsageCsvModal.tsx

```diff
@@ -0,0 +1,76 @@
+import { DatePicker, Form, Modal } from 'antd';
+import dayjs, { Dayjs } from 'dayjs';
+import React from 'react';
+
+import { handlePath, serverAddress } from 'routes/utils';
+
+interface Props {
+  afterDate: Dayjs,
+  beforeDate: Dayjs,
+  onVisibleChange: (visible: boolean) => void;
+}
+
+const ClusterHistoricalUsageCsvModal: React.FC<Props> = (
+  { afterDate, beforeDate, onVisibleChange }: Props,
+) => {
+  const [ form ] = Form.useForm();
+
+  const handleOk = (event: React.MouseEvent): void => {
+    const formAfterDate = form.getFieldValue('afterDate');
+    const formBeforeDate = form.getFieldValue('beforeDate');
+    const searchParams = new URLSearchParams;
+
+    searchParams.append('timestamp_after', formAfterDate.startOf('day').toISOString());
+    searchParams.append('timestamp_before', formBeforeDate.endOf('day').toISOString());
+
+    handlePath(event, {
+      external: true,
+      path: serverAddress('/resources/allocation/raw?' + searchParams.toString()),
+      popout: true,
```

Comment:
**question:** shouldn't `popout` be `false` so the download doesn't pop out a new tab? I took a quick peek at `Download Model` and the link there doesn't specify a `popout` of `true` either.